src/Context/RichSnippetContext.php

```diff
@@ -23,7 +23,7 @@ final class RichSnippetContext
     public function __construct(
         RequestStack $requestStack,
         iterable $subjectFetchers,
-        iterable $richSnippetFactories
+        iterable $richSnippetFactories,
     ) {
         $this->request = $requestStack->getMainRequest();
```

Comment:
```suggestion
        $this->request = $requestStack->getCurrentRequest();
```
IMO, we need also change it ;)

Reply:
If I'm not mistaken, this is intentional. 

The CurrentRequest might be a subrequest, for example during the rendering of a template or a controller route. 

The MainRequest will always be the browser triggered request, which we need to identify the RichSnippet subject. 

Does it make sense ? Do you have examples where it is better to use the CurrentRequest ?

Reply:
Ok, i see.